mlflow/pyfunc/scoring_server/client.py

```diff
@@ -2,6 +2,19 @@
 import time
 from mlflow.pyfunc import scoring_server
 import json
+import datetime
+
+# Extended `DateTimeEncoder` in https://stackoverflow.com/a/12126976 to support `pandas.TimeStamp`
+class DateTimeEncoder(json.JSONEncoder):
```

Comment:
Rename the class name to be "CustomJsonEncoder" ?
The name `DateTimeEncoder` looks like it only support DataTime type but actually it supports other basic pandas data types.

Reply:
> it also supports other basic pandas data types.

It doesn't.

Reply:
uh ?
if I understand correctly,
the overridden `default(self, o):`
it calls `super(DateTimeEncoder, self).default(o)` on non timestamp related types, this makes it support other pandas types.

Reply:
```python
import datetime
import pandas as pd
import json


class _DateTimeEncoder(json.JSONEncoder):
    def default(self, o):
        import pandas as pd

        if isinstance(o, (datetime.datetime, datetime.date, datetime.time, pd.Timestamp)):
            return o.isoformat()
        elif isinstance(o, datetime.timedelta):
            return (datetime.datetime.min + o).time().isoformat()

        return super().default(o)


df = pd.DataFrame({"col1": [1, 2], "col2": [3, 4]})
print(json.dumps(df, cls=_DateTimeEncoder))
```

results in:

```
Traceback (most recent call last):
  File "a.py", line 19, in <module>
    print(json.dumps(df, cls=_DateTimeEncoder))
  File "/home/haru/miniconda3/envs/mlflow-dev-env/lib/python3.7/json/__init__.py", line 238, in dumps
    **kw).encode(obj)
  File "/home/haru/miniconda3/envs/mlflow-dev-env/lib/python3.7/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/home/haru/miniconda3/envs/mlflow-dev-env/lib/python3.7/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "a.py", line 15, in default
    return super().default(o)
  File "/home/haru/miniconda3/envs/mlflow-dev-env/lib/python3.7/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type DataFrame is not JSON serializable
```

Reply:
oh, my typo. I meant it also support other basic python types besides timestamp related types

Reply:
`CustomJsonEncoder` sounds too generic. I prefer `_DateTimeEncoder` because it describes what it can encode.